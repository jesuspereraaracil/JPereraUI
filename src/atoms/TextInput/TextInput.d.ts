import React, { ChangeEventHandler } from 'react';
interface TextInputProps {
    disabled: boolean;
    value?: string;
    onChangeHandler?: ChangeEventHandler;
    placeholder?: string;
}
export declare const TextInput: React.FC<TextInputProps>;
export {};
