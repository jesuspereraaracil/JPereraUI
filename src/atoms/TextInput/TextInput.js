import React from 'react';
import { TextInputStyled } from './TextInput.styled';
export const TextInput = ({ disabled, value, onChangeHandler, placeholder }) => {
    return React.createElement(TextInputStyled, { disabled: disabled, value: value, onChange: onChangeHandler, placeholder: placeholder });
};
