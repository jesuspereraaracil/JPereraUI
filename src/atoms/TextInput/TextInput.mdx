{/* TextInput.mdx */}

import { Canvas, Meta } from '@storybook/blocks';

import * as TextInputStories from './TextInput.stories';

<Meta of={TextInputStories} />

# TextInput

A TextInput is a place to write.

## Default TextInput
This is the default behavior of the TextInput.
<Canvas of={TextInputStories.Default} />

## Error state TextInput
The TextInput has an error state.
<Canvas of={TextInputStories.Error} />

## Disabled state TextInput
The TextInput also has a disabled state
<Canvas of={TextInputStories.Disabled} />
