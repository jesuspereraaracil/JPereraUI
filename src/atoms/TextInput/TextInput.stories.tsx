import type { Meta, StoryObj } from '@storybook/react';
import { TextInput } from './TextInput';

const meta: Meta<typeof TextInput> = {
    /* ðŸ‘‡ The title prop is optional.
     * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
     * to learn how to generate automatic titles
     */
    title: 'TextInput', component: TextInput,
};
export default meta;
type Story = StoryObj<typeof TextInput>;

export const Primary: Story = {
    args: {
        error: false,
    }
};

export const Error: Story = {
    args: {
        error: true,
    }
};
