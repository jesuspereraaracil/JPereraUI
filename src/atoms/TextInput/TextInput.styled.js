import styled from '@emotion/styled';
import { theme } from '../../theme';
export const TextInputStyled = styled.input `
  background-color: ${theme.colors.background};
  border-radius: ${theme.spacing.NORMAL};
  border: 0;
  color: ${theme.colors.font};
  height: ${theme.spacing.HUGE};
  outline: 0;
  padding: ${theme.spacing.SMALL} ${theme.spacing.NORMAL};
  
  &:focus, &:active, &:focus-visible {
    border: 1px solid ${theme.colors.primary};
  }
  
  &:hover {
    box-shadow: 0 0 ${theme.spacing.NORMAL} 0 ${theme.colors.primary};
  }
  
  &:disabled {
    cursor: not-allowed;
  }
`;
