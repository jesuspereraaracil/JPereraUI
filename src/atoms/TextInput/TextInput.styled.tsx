import styled, { css } from 'styled-components';
import { theme } from '../../theme';
import { TextInputProps } from './TextInput.types';

export const TextInputStyled = styled.input<TextInputProps>`
    background-color: ${theme.colors.background};
    border-radius: ${theme.spacing.NORMAL};
    border: 1px solid ${theme.colors.border};
    color: ${theme.colors.font};
    height: ${theme.spacing.HUGE};
    outline: 0;
    padding: ${theme.spacing.SMALL} ${theme.spacing.NORMAL};

    &:focus,
    &:active,
    &:focus-visible {
        border: 1px solid ${theme.colors.primary};
    }

    &:hover {
        box-shadow: 0 0 ${theme.spacing.NORMAL} 0 ${theme.colors.primary};
    }

    ${({ disabled }) =>
        disabled &&
        css`
            background-color: ${theme.colors.backgroundDisabled};
            color: ${theme.colors.fontDisabled};
            cursor: not-allowed;

            &:focus,
            &:active,
            &:focus-visible {
                border: 1px solid ${theme.colors.borderDisabled};
            }

            &:hover {
                box-shadow: 0 0 ${theme.spacing.NORMAL} 0 ${theme.colors.border};
            }
        `}

    ${({ error }) =>
        error &&
        css`
            border: 1px solid ${theme.colors.states.error};

            &:focus,
            &:active,
            &:focus-visible {
                border: 1px solid ${theme.colors.states.error};
            }

            &:hover {
                box-shadow: 0 0 ${theme.spacing.NORMAL} 0
                    ${theme.colors.states.error};
            }
        `}
`;
