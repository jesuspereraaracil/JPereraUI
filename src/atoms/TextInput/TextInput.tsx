import { TextInputStyled } from './TextInput.styled';
import { TextInputProps } from './TextInput.types';
import React from 'react';
import { Container } from '../Container/Container.styled';

export const TextInput = ({
  extraClassnames,
  ...props
}: TextInputProps): React.JSX.Element => (
  <Container>
    <TextInputStyled {...props} className={extraClassnames} />
  </Container>
);
