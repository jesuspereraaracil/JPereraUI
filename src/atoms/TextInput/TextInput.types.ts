import { ChangeEventHandler } from 'react';

export interface TextInputProps {
    disabled?: boolean;
    value?: string;
    onChangeHandler?: ChangeEventHandler;
    placeholder?: string;
    extraClassnames?: string;
    error?: boolean
}
