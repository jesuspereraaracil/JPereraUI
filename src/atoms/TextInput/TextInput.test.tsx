import { fireEvent, render } from '@testing-library/react';
import { TextInput } from './TextInput';

describe('TextInput', () => {
  const TEST_ID = 'TextInputID';
  const RANDOM_TEXT = 'RANDOM_TEXT';
  const EMPTY_TEXT = '';

  const setup = () => {
    return render(<TextInput data-testid={TEST_ID} />);
  };

  test('Should render correctly', () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Should be able to write', () => {
    const { getByTestId } = setup();
    const input = getByTestId(TEST_ID);
    fireEvent.change(input, { target: { value: RANDOM_TEXT } });
    expect((input as HTMLInputElement).value).toBe(RANDOM_TEXT);
    fireEvent.change(input, { target: { value: EMPTY_TEXT } });
    expect((input as HTMLInputElement).value).toBe(EMPTY_TEXT);
  });
});
