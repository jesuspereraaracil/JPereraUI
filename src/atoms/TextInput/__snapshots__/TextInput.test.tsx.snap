// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Should render correctly 1`] = `
.c1 {
  background-color: #F0F0F0;
  border-radius: 8px;
  border: 1px solid #808080;
  box-sizing: border-box;
  color: #202020;
  font-size: 16px;
  outline: 0;
  padding: 4px 8px;
  width: 100%;
}

.c1:focus,
.c1:active,
.c1:focus-visible {
  border: 1px solid #38A9FF;
}

.c1:hover {
  box-shadow: 0 0 8px 0 #38A9FF;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px 8px;
  width: 100%;
}

<div
  class="c0"
>
  <input
    class="c1"
    data-testid="TextInputID"
  />
</div>
`;
